Concepts - first set of rewrites

ljsinclair opened the pull request:
## Please ignore the majority of files changed

The majority of files changed in this PR are:
* minor standardization changes (e.g., column headings)
* structural changes to rearrange files in a better order (e.g., changing the concepts content to sit under two parents)
* bug fixes, broken links etc.

These have been tested and work so don't need anything further.

## Please review JUST the concepts content

You will need to build the site to be able to review the content because there are common include files in play

* [Local build documentation](https://github.com/FeatureBaseDB/featurebase-docs/tree/main/help-on-help/local-build)

The following files are ground-up rewrites and **definitely** need reviewing:

| Built help page | Source page | Description |
|---|---|---|
| https://docs.featurebase.com/docs/concepts/concepts-home/ | /docs/concepts/concepts-home.md | gives some context for cardinality, how normalization works at a STRATOSPHERIC level (as in uber high-level) leading into how FeatureBase doesn't normalize data and pointing users to further information |
| https://docs.featurebase.com/docs/concepts/concept-fb-bitmaps | /docs/concepts/concept-fb-bitmaps.md | parent page, ways cardinality and bitmaps work, and don't work well, leading into... |
| https://docs.featurebase.com/docs/concepts/concept-pt1-bitmap-index |  /docs/concepts/concept-pt1-bitmap-index.md | how bit-slicing works with integer values |
| https://docs.featurebase.com/docs/concepts/concept-pt2-range-encode-bit-slice | /docs/concepts/concept-pt2-range-encode-bit-slice.md | range-encoding the bit sliced data, explaining how that works |
| https://docs.featurebase.com/docs/concepts/concept-ingest-eg-key-cardinality-low | /docs/concepts/concept-ingest-eg-key-cardinality-low.md | super-duper rudimentary source data which demonstrates:<br/>* how the choice of unique identifier/key will change the number of rows in destination FB tables<br/>* how SET data types allow low cardinality data to be ingested by FB<br/>* SELECT syntax used to query the specifics of data in the SET data type column |

## NOTES
* I have undoubtedly got some of the concepts [wrong | very wrong | totally missed the bus] so **please make corrections** because if I haven't got this by now, I possibly never will
* Very soon I'm going to be asking for information on how other data types get converted to base-2 ready for bit slicing :)

Special thanks to whomever wrote the following blog posts:
* https://www.featurebase.com/blog/bitmaps-making-real-time-analytics-real - used to help write the fb-bitmaps content
* https://www.featurebase.com/blog/range-encoded-bitmaps -- used to write the pt1 and pt2 content

ljsinclair commented on docs/concepts/concept-pt2-range-encode-bit-slice.md:
```diff
@@ -0,0 +1,203 @@
+grand_parent: Concepts
+nav_order: 2
+---
+
```
I'm going to admit it now: I suspect this breaks my own rule for endless scroll.
For the moment I want to leave as-is because it needs a review for truth and accuracy.
Post review I'll split it up probably into individual pages based on the "step" headings

ljsinclair replied:
I think with the removal of range encoding this page is (a) better and (b) not so much of an endless scroll.
What do you think?

ljsinclair commented on old-backup/concept-data-modeling.md:
```diff
@@ -0,0 +1,104 @@
+<!-- DO NOT REVIEW -- THIS CONTENT FOR NEXT STAGE
```
Like it says on the tin, this content is for the next PR.
It was written during work on this PR whilst conceptualising information.

ljsinclair commented on old-backup/fb-solutions-denormalized-data.md:
```diff
@@ -0,0 +1,52 @@
+<!-- This content was written originally for the Concepts homepage.
```
Like it says on the tin, this content is for the next PR.
It was written during work on this PR whilst conceptualising information.

ljsinclair commented on old-backup/ingest-method-rewrites.md:
```diff
@@ -0,0 +1,48 @@
+<!--DO NOT REVIEW-->
```
Like it says on the tin, this content is for the next PR.
It was written during work on this PR whilst conceptualising information.

ljsinclair commented on docs/cloud/cloud-databases/cloud-db-serverless-scale.md:
```diff
@@ -19,7 +19,7 @@ This procedure explains how to scale a serverless database.
 {% include /cloud/cloud-before-begin.md %}
 * [Learn about FeatureBase Serverless](/docs/cloud/cloud-databases/cloud-db-serverless-home)
-* [Learn how to create a serverless database](/docs/cloud/cloud-databases/cloud-db-serverless-create)
+* [Learn how to create a serverless database](/docs/cloud/cloud-databases/cloud-db-create-serverless)
```
broken link fix

ljsinclair commented on docs/sql-guide/statements/statement-insert.md:
````diff
@@ -106,10 +106,24 @@ idsetcolq idsetq timequantum 'YMD'
 VALUES (1, {'2018-08-31T00:00:00Z', ['A','B']}, {1676649734, [1]});
 ```
 
+### CREATE TABLE with `STRINGSET` data types
````
This content came about because of the work on the new concept example.
There wasn't an example of inserting into a stringset data type so here it is

ljsinclair commented on docs/sql-guide/statements/statements-home.md:
```diff
@@ -47,10 +47,10 @@ SQL statements can be executed in the Query editor.
 ## System statements
 
-* [KILL QUERY statement](/docs/sql-guide/statements/statement-kill)
+* [KILL QUERY statement](/docs/sql-guide/statements/statement-kill-query)
```
broken link fix

ljsinclair commented on _includes/sql-guide/table-create-eg-stringset-datatype.md:
````diff
@@ -0,0 +1,7 @@
+```sql
````
created for new concept example, plus ended up as example table for insert into stringset which didn't exist

ljsinclair commented on docs/cloud/cloud-tables/cloud-table-manage.md:
```diff
@@ -36,7 +36,7 @@ Perform data modeling **before** creating tables to avoid issues.
 ## Column data types and constraints
 
-* [Learn about data types and constraints](/docs/sql-preview/data-types/data-types-home)
+* [Learn about data types and constraints](/docs/sql-guide/data-types/data-types-home)
```
I genuinely don't know how this stayed hidden so long.

gthrone25 commented on docs/concepts/concept-ingest-eg-large-dataset.md:
```diff
@@ -1,9 +1,9 @@
 ---
 title: A True Crime Story
```
Not sure I understand this rename. This is not a large dataset, and we are still keeping the same title?

ljsinclair replied:
The file rename was to give context in relation to the new examples coming in.
The file title is staying the same because I haven't had a chance yet to perform rewrites and recontextualisation on the content.
re, large dataset, that's stated in the text: "...You realize this data is way too big to analyze on your local/virtual workspace..."
Happy to rename it to something else.

gthrone25 commented on docs/concepts/concept-pt1-bitmap-index.md:
```diff
@@ -0,0 +1,135 @@
+# Part 1 - bitmap indexes
+{: .no_toc }
+
+This is the first in a series of articles that explains how FeatureBase uses bitmaps, bit-slicing and range-encoding to:
```
no range encoding. Not commenting on every instance of it, but it will need to be removed

ljsinclair replied:
removed

gthrone25 reviewed (commented):
@ljsinclair noticed a couple of things I commented on. Up to you on what to address but the biggest is that we don't use range encoding. Progress over perfection, so up to you

ljsinclair commented on _includes/concepts/concept-range-encoding-summary.md:
```diff
@@ -0,0 +1,5 @@
+When range encoding, all values greater than a specific value are set to 1.
```
That's great to know because this was really confusing.

ljsinclair replied:
removed and fixed

ljsinclair commented on docs/concepts/concept-examples.md:
```diff
@@ -0,0 +1,36 @@
+* [Learn how data is encoded in bitmap indexes](/docs/concepts/concept-fb-bitmaps)
+* [Learn how to structure your data ready to import to FeatureBase](/docs/concepts/overview-data-modeling)
+
+## SQL `BULK INSERT`
```
This is why we test :)

ljsinclair replied:
fixed

ljsinclair commented on docs/concepts/concept-examples.md:
```diff
@@ -0,0 +1,36 @@
+nav_order: 3
+---
+
+# Example data and import methods
```
Well they don't literally live there, there's just links. And the examples demonstrate how different import/ingest methods work.

The examples will be built out based on Jacob's WIP file which will add to these and give additional context on specifics. 

Ideally the ingest examples would also be rejigged with the schema and data Jacob put together so there's consistency throughout.

ljsinclair commented on docs/concepts/concept-fb-bitmaps.md:
```diff
@@ -0,0 +1,12 @@
+---
+title: Bitmap index overview
```
It's needed as a parent for 2x parts.
Have added context based on intro to both parts and some additional rewrites

ljsinclair commented on docs/concepts/concept-pt2-range-encode-bit-slice.md:
```diff
@@ -0,0 +1,203 @@
+* one bitmap representing `1`
+* one bitmap representing `0`
+
+### Bitmap `2^0 = 1`
```
Deleted :)

ljsinclair commented on docs/concepts/concepts-home.md:
```diff
@@ -3,9 +3,94 @@ title: Concepts
 nav_order: 2
-has_toc: true
 ---
+# How does FeatureBase differ to a traditional database?
```
Well the intent here was to put together a stratospheric-level overview of concepts regarding how data is treated in trad databases, all of which are then relevant to the content that follows.

And if I've missed anything please let me know :)

ljsinclair commented:
Hi @gthrone25 > have addressed your comments (thanks for these, esp the range encoding stuff which was definitely not great).
Let me know if this is ok, and if you're happy one of us can rebase and all will be peachy with the universe :D
